tests/foreman/api/test_bookmarks.py

```diff
@@ -22,364 +24,356 @@
+        2. List the bookmarks
+
+    :expectedresults: No errors, Bookmark is listed, controller matches the
+        controller the bookmark was created for
+
+    :CaseImportance: Critical
+    """
+    query = random.choice(list(valid_data_list().values()))
+    bm = entities.Bookmark(controller=controller, query=query).create()
+    assert bm.controller == controller
+    assert bm.query == query
+
+
+@pytest.mark.tier1
+@pytest.mark.parametrize('public', (True, False))
+@pytest.mark.parametrize('controller', CONTROLLERS)
+def test_positive_create_public(controller, public):
+    """Create a public bookmark
+
+    :id: 511b9bcf-0661-4e44-b1bc-475a1c207aa9
+
+    :parametrized: yes
+
+    :Steps:
+
+        1. Create a bookmark with a random name and public = true
+        2. List the bookmarks
+
+    :expectedresults: No errors, Bookmark is listed, controller matches the
+        entity the bookmark was created for and is displayed as public
+
+    :CaseImportance: Critical
+    """
+    bm = entities.Bookmark(controller=controller, public=public).create()
+    assert bm.controller == controller
+    assert bm.public == public
+
+
+@pytest.mark.tier1
+@pytest.mark.parametrize('controller', CONTROLLERS)
+def test_negative_create_with_invalid_name(controller):
+    """Create a bookmark with invalid name
+
+    :id: 9a79c561-8225-43fc-8ec7-b6858e9665e2
+
+    :parametrized: yes
+
+    :Steps:
+
+        1. Attempt to create a bookmark with providing an invalid name
+        2. List the bookmarks
+
+    :expectedresults: Error returned, Bookmark is not created (not listed)
+
+    :CaseImportance: Critical
+    """
+    name = random.choice(invalid_values_list())
+    with pytest.raises(HTTPError):
+        entities.Bookmark(controller=controller, name=name, public=False).create()
+    result = entities.Bookmark().search(query={'search': f'name="{name}"'})
+    assert len(result) == 0
+
+
+@pytest.mark.tier1
+@pytest.mark.parametrize('controller', CONTROLLERS)
+def test_negative_create_empty_query(controller):
+    """Create a bookmark with empty query
+
+    :id: 674d569f-6f86-43ba-b9cc-f43e05e8ab1c
+
+    :parametrized: yes
+
+    :Steps:
+
+        1. Create a bookmark with providing an empty query
+        2. List the bookmarks
+
+    :expectedresults: Error notification - search query cannot be empty,
+        Bookmark is not created (not listed)
+
+    :CaseImportance: Critical
+    """
+    name = gen_string('alpha')
+    with pytest.raises(HTTPError):
+        entities.Bookmark(controller=controller, name=name, query='').create()
+    result = entities.Bookmark().search(query={'search': f'name="{name}"'})
+    assert len(result) == 0
+
+
+@pytest.mark.tier1
+@pytest.mark.parametrize('controller', CONTROLLERS)
+def test_negative_create_same_name(controller):
+    """Create bookmarks with the same names
+
+    :id: f78f6e97-da77-4a61-95c2-622c439d325d
+
+    :parametrized: yes
+
+    :Setup: Create a bookmark with a random name
```

Comment:
I think no need to mention setup separately here, we can include it in test steps.

Reply:
Updated to remove `:Setup:` tokens and move them into `:Steps:`, since they all take place within the test method and not in setup fixtures.